backend/inspirehep/records/api/literature.py

```diff
@@ -89,6 +89,53 @@ def create(
             record.push_authors_phonetic_blocks_to_redis()
             return record
 
+    @classmethod
+    def get_es_linked_records_from_dict_field(cls, data, path):
+        """Return the generator of linked records from specified path for records in ES.
+
+        Args:
+            data (dict): data from which records should be extracted
+            path (str): the path of the linked records.
+        Yields:
+            InspireRecord: the linked records.
+        Examples:
+            >>> data = {
+                'references': [
+                    {
+                        'record': {
+                            '$ref': 'http://localhost/literature/1'
+                        }
+                    }
+                ]
+            }
+            >>>  records = LiteratureRecord.get_linked_records_from_field(
+                data, "references.record")
+            >>> list(records)
+            [
+                {
+                    '$schema': 'http://localhost/schemas/records/hep.json'
+                    'control_number': 1,
+                    'authors': [(...)],
+                    'arxiv_eprints':  [(...)],
+                    'collaborations': [(...)],
+                }
+            ]
+        """
+        pids = cls._get_linked_pids_from_field(data, path)
+        if pids:
+            results = LiteratureSearch.get_records_by_pids(pids)
+            for result in results:
+                try:
+                    rec_data = json.loads(result._ui_display)
+                except AttributeError:
+                    LOGGER.exception(
+                        "Record do not have _ui_display field!", record=result.meta.id
+                    )
+                    continue
+                result._d_.update(rec_data)
+                yield InspireRecord.get_class_for_record(result._d_)(result._d_)
```

Comment:
* Isn't this always a `LiteratureRecord` given you're doing a `LiteratureSearch`?
* I know the semantics of this in invenio are not very clear (you can have a `Record` that's not backed by a model instance) but why should this `LiteratureRecord`?